core/src/main/scala/zio/logging/LogAnnotation.scala

```diff
@@ -8,6 +8,9 @@ import scala.reflect.ClassTag
  */
 final case class LogAnnotation[A: ClassTag](name: String, initialValue: A, combine: (A, A) => A, render: A => String) {
   self =>
+
+  def apply(value: A): LogContext => LogContext = _.annotate(self, combine(initialValue, value))
```

Comment:
Would be nice to use this in some code to do cleanup. There are a few places that could benefit.

Reply:
I actually dropped `locallyAnnotate` because with this I don't think we need it. I updated examples and docs accordingly.